### ISSUE TITLE
Warrior - Last Stand (BB #552)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#552]
This issue was migrated from bitbucket.
**Original Reporter:** smoldar
**Original Date:** 15.03.2014 11:56:38 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552

<hr>

This spell bugging character, after end of Last Stand, player will be casted as feign death, after he is RESSURECTED.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 15.03.2014 12:14:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9137845

<hr>

When you will have hp under 30% with last stand. After of removing last stand you will lost 30% of your hp = you will be death, but after 1-2sec you will be ressurected instantly.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 15.03.2014 15:07:32 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9139298

<hr>

It doesn't seem to revive me instantly but it does bug me out. I still get attacked and its like im not dead. But I can't move and I can release spirit.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 18.03.2014 07:15:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9181706

<hr>

I think if someone will have under 30% hp and after lost aura Last Stand, it may set hp to 1 or die... i dont know now

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 18.03.2014 08:25:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9182629

<hr>

You're on the right track. I have a test fix for this if you could try it
when I post tomorrow that'd be great

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 20.03.2014 12:23:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9227132

<hr>

Can you post this fix ? I will try it :) thanks

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 20.03.2014 12:37:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9227355

<hr>

Issue #562 was marked as a duplicate of this issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 20.03.2014 16:24:19 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9231734

<hr>

Unit.cpp

```
#!c++

@@ -9995,6 +9995,8 @@
 {
     if (getDeathState() == JUST_DIED)
         val = 0;
+    else if (GetTypeId() == TYPEID_PLAYER && getDeathState() == DEAD)
+        val = 1;
     else
     {
         uint32 maxHealth = GetMaxHealth();
@@ -10024,6 +10026,9 @@

 void Unit::SetMaxHealth(uint32 val)
 {
+    if (!val)
+        val = 1;
+
     float healthPct = GetMaxHealth();
     SetUInt32Value(UNIT_FIELD_MAXHEALTH, val);
```

SpellAuras.cpp

```
#!c++

@@ -2287,7 +2287,6 @@
                 }
                 return;
             }
-
             break;
         }
         case SPELLFAMILY_MAGE:
@@ -4780,4 +4779,6 @@
             if (int32(m_target->GetHealth()) > m_modifier.m_amount)
                 m_target->ModifyHealth(-m_modifier.m_amount);
             else
+            {
+                m_target->setDeathState(JUST_DIED);
                 m_target->SetHealth(1);
@@ -4783,4 +4784,5 @@
                 m_target->SetHealth(1);
+            }
             m_target->HandleStatModifier(UNIT_MOD_HEALTH, TOTAL_VALUE, float(GetModifierValue()), apply);
         }
     }
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 20.03.2014 18:20:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9233624

<hr>

I tested this, this kill warrior. If warrior have about 1-29% hp, buff last stand will be expired, after warrior will die. If warrior have more than 30% all is ok :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 20.03.2014 19:45:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9235182

<hr>

Thanks. I assume this is how the spell should work? I'll commit this
tonight then. Thanks for posting this issue and testing the solution.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 21.03.2014 07:37:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9242018

<hr>

Works better, but if you lost buff when you have hp between 1-30%, you will die, i think you must get hp to 1, not die

Tested other classes, priest with form and other dying types for prevention of creation new bugs. :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 21.03.2014 18:32:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/552/#comment-9252358

<hr>

[Core/Spells] Fix issues with spells like Last Stand. Closes issue #552

→ <<cset 82abddd00061>>
